### PR TITLE
Port channel-based bot messages over to BotChannel

### PR DESCRIPTION
**Pull request by bashtech**
This PR converts all of the channel-based messages to BotChannel. DMs and reactions haven't been updated yet.

Please make sure the environment variable, BOT_API_BASE_URL, is defined before deploying since the hard-coded value was removed.

### CODE REVIEW

**Comment by flashadvocate** [comment]
@bashtech Can you update `.env.example` with the new environment variable?

File: app/Channels/Messages/BotMessage.php [hunk 1]
```diff
@@ -113,14 +130,15 @@ public function send(): array
          */
         return [
             'embeds' => [[
-                'color' => $this->color ?? 0,
-                'description' => $this->message,
+                'co2lor' => $this->color ?? 0,
```
Comment (flashadvocate) [inline app/Channels/Messages/BotMessage.php:133]: Think this was a typo

```suggest
'color' => $this->color ?? 0,
```

**Comment by bashtech** [comment]
@flashadvocate  It's already defined in `.env.example`. 

https://github.com/ClanAODDev/tracker_v3/blob/e30753bff394ae9f10cf90ee446eb24777c7b84d/.env.example#L38

File: app/Channels/Messages/BotMessage.php (hunk 1 above)
Comment (bashtech) [reply app/Channels/Messages/BotMessage.php]: Fixed in 60684353de3ac53f6d5e57f435d46d7d44c2b679

File: app/Channels/Messages/BotMessage.php [hunk 2]
```diff
@@ -113,14 +130,15 @@ public function send(): array
...
+                'description' => $this->message ?? "",
                 'author' => [
                     'name' => $this->title,
                     'icon_url' => 'http://tracker.clanaod.net/images/logo_v2.png',
                     'url' => $this->url ?? 'https://tracker.clanaod.net',
                 ],
                 'fields' => $this->fields ?? [],
+                'id' => $this->messageId,
```
[the remark this answers is not in the document]
Comment (flashadvocate) [reply app/Channels/Messages/BotMessage.php:141]: I think if we drop this attribute, this PR should be safe to merge

[the remark this answers is not in the document]
Comment (bashtech) [reply app/Channels/Messages/BotMessage.php]: It was copied from the DiscordMessage class, but it is in the wrong location anyway. I'll remove it for now.

https://github.com/ClanAODDev/tracker_v3/blob/e30753bff394ae9f10cf90ee446eb24777c7b84d/app/Channels/Messages/DiscordMessage.php#L125

**Pull request merged**